tinygrad/runtime/ops_python.py

```diff
@@ -68,11 +68,11 @@ def __call__(self, *bufs, global_size:Tuple[int,int,int]=(1,1,1), local_size:Tup
         dl[i] = dtype
         if uop is Ops.DEFINE_GLOBAL:
           assert dtype.fmt is not None
-          ul[i] = [pbufs.pop(0).cast(dtype.fmt)] * warp_size
+          ul[i] = [pbufs.pop(0).cast(cast(LegacyFmtStrType, dtype.fmt))] * warp_size  # TODO typeshed missing stub for cast
```

Comment:
The typeshed stubs for `memoryview.cast` have not yet been updated for half  ([mypy/typeshed/stdlib/builtins.pyi](https://github.com/python/mypy/blob/v1.13.0/mypy/typeshed/stdlib/builtins.pyi#L769)), so the Literal type is narrowed for < python 3.12 support.  (https://docs.python.org/3/library/typing.html#typing.cast).

The alternative is to perform runtime checks on `sys.version_info < (3,12)` and then throwing asserts if `dtype.fmt` corresponds to half ('e', '@e').

Reply:
If you are going to cast it anyway, can it just be a string?